include/protozero/iterators.hpp

```diff
@@ -216,8 +217,73 @@ class const_fixed_iterator {
         return !(*this == rhs);
     }
 
+    const_fixed_iterator& operator--() {
+        m_data -= sizeof(value_type);
+        return *this;
+    }
+
+    const_fixed_iterator operator--(int) {
+        const const_fixed_iterator tmp{*this};
+        --(*this);
+        return tmp;
+    }
+    
+    friend bool operator<(const const_fixed_iterator& lhs, const const_fixed_iterator& rhs) {
+        return (rhs.m_data - lhs.m_data) > 0;
+    }
+
+    friend bool operator>(const const_fixed_iterator& lhs, const const_fixed_iterator& rhs) {
+        return rhs < lhs;
+    }
+
+    friend bool operator<=(const const_fixed_iterator& lhs, const const_fixed_iterator& rhs) {
+        return !(lhs > rhs);
+    }
+
+    friend bool operator>=(const const_fixed_iterator& lhs, const const_fixed_iterator& rhs) {
+        return !(lhs < rhs);
+
+    }
+
+    const_fixed_iterator& operator+=(difference_type val) {
+        m_data += (sizeof(value_type) * val);
+        return *this;
+    }
+
+    friend const_fixed_iterator operator+(const const_fixed_iterator& lhs, difference_type rhs) {
+        const_fixed_iterator tmp{lhs};
+        tmp.m_data += (sizeof(value_type) * rhs);
+        return tmp;
+    }
+
+    friend const_fixed_iterator operator+(difference_type lhs, const const_fixed_iterator& rhs) {
+        const_fixed_iterator tmp{rhs};
+        tmp.m_data += (sizeof(value_type) * lhs);
+        return tmp;
+    }
+
+    const_fixed_iterator& operator-=(difference_type val) {
+        m_data -= (sizeof(value_type) * val);
+        return *this;
+    }
+
+    friend const_fixed_iterator operator-(const const_fixed_iterator& lhs, difference_type rhs) {
+        const_fixed_iterator tmp{lhs};
+        tmp.m_data -= (sizeof(value_type) * rhs);
+        return tmp;
+    }
+
+    friend difference_type operator-(const const_fixed_iterator& lhs, const const_fixed_iterator& rhs) {
+        return static_cast<difference_type>(lhs.m_data - rhs.m_data) / static_cast<difference_type>(sizeof(T));
+    }
+
+    reference operator[](difference_type n) const {
```

Comment:
This must return `value_type`, not a reference! Otherwise we'll return a reference to a local, because `operator*()` also returns a value.